### ISSUE TITLE
jax.ad_checkpoint package does not appear in API docs

### ISSUE THREAD

**Issue by shoyer** [jax-ml/jax#33630]
It's mentioned in the gradient checkpointing docs (https://docs.jax.dev/en/latest/gradient-checkpointing.html), so it's clearly a public API, but at least some of the exported functions (e.g., `checkpoint_name`) are not separately documented, as is standard for JAX.

**Comment by jakevdp** [comment]
Thanks! I'll take a look.

**Comment by jakevdp** [comment]
@shoyer proposed fix in #33634

I didn't add anything beyond `checkpoint` and `remat`, because nothing else currently has a docstring. Are there other APIs that you think should be in the HTML docs?

**Comment by shoyer** [comment]
To be honest, it's a little unclear to me what is supposed to be a public API in `jax.ad_checkpoint`, or whether it's really worth keeping it around at all.

- `checkpoint_name` is the one obvious example that should be documented, given that it is required for some checkpoint policies.
- `checkpoint`/`remat` and `checkpoint_policies` are already exposed in the top level `jax` namespace.
- `print_saved_residuals` looks like an internal utility.
- `Recompute`/`Savable`/`Offloadable` seem to be necessary for writing custom checkpointing policies, but that API is not documented as far as I can tell.

**Comment by jakevdp** [comment]
Actually, it looks like `jax.ad_checkpoint.checkpoint` is a different function than `jax.checkpoint`. I'm not clear on why you would use one versus the other.

**Comment by jakevdp** [comment]
Oh, I see the API at `jax.checkpoint` has a `concrete` argument related to the deprecation in https://docs.jax.dev/en/latest/jep/11830-new-remat-checkpoint.html. We could probably remove that.

I think it's probably best for us to deprecate the aliases in `jax.ad_checkpoint` and prefer the APIs in the `jax` namespace, since they will be more used.
